text/0001-data-collector.md

```diff
@@ -0,0 +1,193 @@
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+`data_collector.py` run inside a Docker container, polling asychronously a fixed list of nodes each second. The endpoint to be polled is 
+
+> GET `/v1a/status`
+
+For each request, part of the response is sent to `network-data-aggregator`:
+
+**Response from `hathor-core`:**
+```
+{
+  "server": {
+    "id": string,
+    "app_version": string,
+    "state": string,
+    "network": string,
+    "uptime": float,
+    "entrypoints": [string, ...]
+  },
+  "peers_whitelist": [string, ...],
+  "known_peers": [
+    {
+      "id": string,
+      "entrypoints": [string, ...],
+      "flags": [string, ...]
+    },
+    ...
+  ],
+  "connections": {
+    "connected_peers": [
+      {
+        "id": string,
+        "app_version": string,
+        "uptime": float,
+        "address": string,
+        "state": string,
+        "last_message": float,
+        "plugins": {
+          "node-sync-timestamp": {
+            "latest_timestamp": int,
+            "synced_timestamp": int
+          }
+        },
+        "warning_flags": [ string, ...]
+      },
+      ...
+    ],
+    "handshaking_peers": [],
+    "connecting_peers": [
+      {
+        "deferred": string,
+        "address": string
+      },
+      ...
+    ]
+  },
+  "dag": {
+    "first_timestamp": int,
+    "latest_timestamp": int
+  }
+}
+```
+
+**Data sent to `network-data-aggregator`:**
+```
+{
+  "server": {
+    "id": string,
+    "app_version": string,
+    "state": string,
+    "network": string,
+    "uptime": float,
+    "entrypoints": [string, ...]
+  },
+  "known_peers": [string, ...],
+  "connected_peers": [
+    {
+      "id": string,
+      "app_version": string,
+      "uptime": float,
+      "address": string,
+      "state": string,
+      "last_message": float,
+      "latest_timestamp": int,
+      "sync_timestamp": int
+      "warning_flags": [ string, ...]
+    },
+    ...
+  ]
+}
+```
+
+`network-data-aggregator` then saves it into Redis under the following keys:
+
+> `'hathor-explorer-service.network.${server.id}'`
+
+**Endpoiont:**
+
+GET `/v1a/network/`
+
+`network-handler` return a list of available nodes PoV (Redis keys)
+
+```
+[string, ...]
+```
+
+**Endpoiont:**
+
+GET `/v1a/network/:hash`
+
+`network-handler` fecth data of given node from Redis and return 
+
+```
+{
+  "server": {
+    "id": string,
+    "app_version": string,
+    "state": string,
+    "network": string,
+    "uptime": float,
+    "entrypoints": [string, ...]
+  },
+  "known_peers": [string, ...],
+  "connected_peers": [
+    {
+      "id": string,
+      "app_version": string,
+      "uptime": float,
+      "address": string,
+      "state": string,
+      "last_message": float,
+      "latest_timestamp": int,
+      "sync_timestamp": int
+      "warning_flags": [ string, ...]
+    },
+    ...
+  ]
+}
+
+```
+
+*Note:* If a invalid hash is sent, 404 is returned.
+
```

Comment:
With this RFC we can get all the nodes saved in redis and the data of each node, right? We still can't get the aggregated data, right?

Reply:
Yes, aggregated network data is coming next.